src/lib/route-broadcaster.js

```diff
@@ -55,30 +55,51 @@ class RouteBroadcaster {
         throw e
       }
     }
-    setInterval(() => this.routingTables.removeExpiredRoutes(), this.routeCleanupInterval)
-    defer.setInterval(function * () {
-      yield this.reloadLocalRoutes()
-      this.broadcast()
-    }.bind(this), this.routeBroadcastInterval)
+    this.removeExpiredRoutesSoon()
+    this.broadcastSoon()
   }
 
-  broadcast () {
-    const adjacentLedgers = Object.keys(this.peersByLedger)
-    const routes = this.routingTables.toJSON(SIMPLIFY_POINTS)
-    for (let adjacentLedger of adjacentLedgers) {
-      const ledgerRoutes = routes.filter((route) => route.source_ledger === adjacentLedger)
+  removeExpiredRoutesSoon () {
+    setTimeout(() => {
       try {
-        this._broadcastToLedger(adjacentLedger, ledgerRoutes)
+        this.routingTables.removeExpiredRoutes()
       } catch (err) {
-        log.warn('broadcasting routes on ledger ' + adjacentLedger + ' failed')
+        log.warn('removing expired routes failed')
         log.debug(err)
       }
-    }
+      this.removeExpiredRoutesSoon()
+    }, this.routeCleanupInterval)
+  }
+
+  broadcastSoon () {
+    defer.setTimeout(function * () {
+      try {
+        yield this.reloadLocalRoutes()
```

Comment:
> I would imagine that the expiry interval would be several times smaller than the broadcast interval, because the former is much cheaper.

In that case, does it make sense to run `removeExpiredRoutes` in this loop as well? Otherwise, we might broadcast already expired routes, right?